src/System.Memory/tests/Span/CopyTo.cs

```diff
@@ -181,6 +182,59 @@ public static void CopyToShorterArray()
             Assert.Equal<int>(expected, dst);  // CopyTo() checks for sufficient space before doing any copying.
         }
 
+        private static void CopyToLargeSizeTestHelper(long BufferSize)
+        {
+            int GuidCount = (int)(BufferSize / Unsafe.SizeOf<Guid>());
+
+            unsafe
+            {
+                bool allocatedFirst = AllocationHelper.TryAllocNative((IntPtr)BufferSize, out IntPtr memBlockFirst);
+                bool allocatedSecond = AllocationHelper.TryAllocNative((IntPtr)BufferSize, out IntPtr memBlockSecond);
+
+                try
+                {
+                    if (allocatedFirst && allocatedSecond)
+                    {
+                        ref Guid memoryFirst = ref Unsafe.AsRef<Guid>(memBlockFirst.ToPointer());
+                        var spanFirst = new Span<Guid>(memBlockFirst.ToPointer(), GuidCount);
+
+                        ref Guid memorySecond = ref Unsafe.AsRef<Guid>(memBlockSecond.ToPointer());
+                        var spanSecond = new Span<Guid>(memBlockSecond.ToPointer(), GuidCount);
+
+                        Guid theGuid = Guid.Parse("900DBAD9-00DB-AD90-00DB-AD900DBADBAD");
+                        for (int count = 0; count < GuidCount; ++count)
+                        {
+                            Unsafe.Add(ref memoryFirst, count) = theGuid;
+                        }
+
+                        spanFirst.CopyTo(spanSecond);
+
+                        for (int count = 0; count < GuidCount; ++count)
+                        {
+                            var guidfirst = Unsafe.Add(ref memoryFirst, count);
+                            var guidSecond = Unsafe.Add(ref memorySecond, count);
+                            Assert.Equal(guidfirst, guidSecond);
+                        }
+                    }
+                }
+                finally
+                {
+                    if (allocatedFirst) AllocationHelper.ReleaseNative(ref memBlockFirst);
+                    if (allocatedSecond) AllocationHelper.ReleaseNative(ref memBlockSecond);
+                }
+            }
+        }
+
+        [Fact]
+        [OuterLoop]
+        [PlatformSpecific(TestPlatforms.Windows | TestPlatforms.OSX)]
+        public static void CopyToLargeSizeTest()
+        {
+            const long TwoGB = 2L * 1024L * 1024L * 1024L;
+            const long TwoGBAnd256Bytes = TwoGB + 256;
 
+            CopyToLargeSizeTestHelper(TwoGB);
+            CopyToLargeSizeTestHelper(TwoGBAnd256Bytes);
```

Comment:
Looks good to me, but can you please add a comment explaining why we are testing size TwoGB + 256 specifically, so it is clear.